### DISCUSSION TITLE
ContactEventsDemo OnRemove and Touching?

### DISCUSSION THREAD

**Discussion by hawkerm** [bepu/bepuphysics2#130] [Q&A] [answered]
Been using physics engines in the past which have kept track of collisions and had events built-in. It's been a bit hard to follow the `ContactEventsDemo` and figure out how to add additional events.

I think one of the current OnContactAdded is actually the touching event, correct? (When the collision still exists and is just updated.)

I think OnRemoved should go in the Flush method somewhere, but seems like the same amount of info isn't available.

**Comment by RossNordby** [comment, marked as answer]
I updated the [demo](https://github.com/bepu/bepuphysics2/blob/master/Demos/Demos/ContactEventsDemo.cs); it now has OnContactAdded, OnContactRemoved, OnStartedTouching, OnTouching, OnStoppedTouching, OnPairCreated, OnPairUpdated, and OnPairEnded.

**Reply by hawkerm** [reply]
Thanks!

**Reply by hawkerm** [reply]
Just tried applying the updates. Had to remove the default interfaces as they don't work in .NET Standard 2.0.

But it also looks like you use a new feature (BodyReference having a CollidableReference) that's not available in the released package 2.3 yet. Is there a planned update dropping soon?

**Reply by RossNordby** [reply]
Not a 'stable' release, no, but there are 2.4.0 beta nuget packages uploaded as prereleases (tick the "Include prerelease" checkbox while searching for nugets).
